### PR TITLE
envoy@1.17 1.17.3 (new formula)

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This adds envoy@1.17 and backfills an alias for 1.18, as that's the current version.

Notably, [Envoy 1.18 changed the config format](https://github.com/envoyproxy/envoy/blob/713b2a6f2b757ff4bea3c51cf89d88f02f56184c/docs/root/version_history/v1.18.0.rst) in an incompatible way with 1.17 which is supported until next year

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

`cd Aliases;ln -s ../Formula/envoy.rb 'envoy@1.18'`

### CODE REVIEW

**Comment by carlocab** [comment]
Why do we need this alias? The alias will need to change when Envoy is upgraded to 1.19, so anyone using `brew install envoy@1.18` will just get an error when that happens.

**Comment by codefromthecrypt** [comment]
good q. minors are usually not compatible in envoy though what isn't incompatible may not effect everyone.

For example, the [pending incompatibilities of 1.19](https://github.com/envoyproxy/envoy/blob/713b2a6f2b757ff4bea3c51cf89d88f02f56184c/docs/root/version_history/current.rst#1190-pending) aren't as shocking as [1.18 which dropped a config format](https://github.com/envoyproxy/envoy/blob/713b2a6f2b757ff4bea3c51cf89d88f02f56184c/docs/root/version_history/v1.18.0.rst)

Knowing this, I'll leave it up to your judgement as I agree alias maintenance is added!

**Comment by carlocab** [comment]
It's not really a question of alias maintenance. I just don't see how this alias will be useful. Or do you think `brew install envoy@1.18` returning an error when Envoy is upgraded to 1.19 would be useful?

**Comment by codefromthecrypt** [comment]
TL;DR; I think most won't need this alias, and only some developers would want to pin to a minor. The problem is there is no packaging for OS/x except homebrew at the moment.

A developer of wasm (filter plugins) may want to know the version is what they think it is and won't care much about the patch version (though sometimes they also break). This has to do with the features changing fast in envoy and that there are a few non-EOL versions https://github.com/envoyproxy/envoy/blob/main/RELEASES.md#release-schedule

I see what you are saying that basically no one can rely on `envoy@1.18` for very long so possibly it is not worth doing.. that they should either not use homebrew to get a known minor, or go straight to the ghcr.io registry?

If there's a way to retain the existing last `envoy@1.18` without copy/pasting the formula, that would be best, but I don't know that option and only know the alias thing or copy/paste. If there's another way all 👂 s

**Comment by carlocab** [comment]
When Envoy 1.19 is released, will Envoy 1.18 continue to receive updates? If it will, and it also meets the rest of the criteria at https://docs.brew.sh/Versions#acceptable-versioned-formulae, then it may be possible to add an `envoy@1.18` formula when 1.19 is released.

**Comment by codefromthecrypt** [comment]
early July 1.19 will be available and a few others including 1.18 will be non-EOL, which means patch releases. So probably the worst case is for a week there could be 4 non-EOL versions out there, but since we are starting with 1.18 and usually one ages out, typically it would be 3 current non-EOL versions and of course, here we could decide to do less.

**Comment by codefromthecrypt** [comment]
probably the bigger point on https://docs.brew.sh/Versions#acceptable-versioned-formulae is "large number of people" frankly I don't think that's the case now, especially vs something like ruby.

I suppose some stats could show that's the case or not, but I'm guessing this might be a reason to close this until it becomes more popular...

**Comment by carlocab** [comment]
A "large number of people" isn't _that_ demanding a requirement. At the moment it amounts to ~800 installs a year. If it seems likely to cross that, then we could probably have an `envoy@1.18` formula. All the earlier versions probably won't qualify, though.

That said, it's still not clear adding this alias now is still useful. Even if we merge this, and I do `brew install envoy@1.18` now, I will still get Envoy 1.19 when I do `brew upgrade` (when Envoy 1.19 is merged here).

**Comment by codefromthecrypt** [comment]
I think the primary use case for `brew install envoy@1.18` is CI matrix or ad-hoc switching back to the prior version because something busted or you were supporting something and there was some doubt about the version being implicated.

**Comment by carlocab** [comment]
Yes. And that'll be possible if/when there is an `envoy@1.18` formula. I don't think it's a good idea to have users relying on its existence until the PR that adds one is accepted.

**Comment by codefromthecrypt** [comment]
gotcha. so basically this is the wrong solution to the problem. Do you prefer a different PR to change the implementation or keep this one? I can see value in the comments here so inclined to change the commits instead of open+ref, but up to you

**Comment by SMillerDev** [comment]
You can just update this one

**Comment by codefromthecrypt** [comment]
OK I updated this to backfill only one previous version 1.17. This is important because its config version is still supported until next year. Hope this is ok, but please tell me if there are any issues with it*

* I noticed many formulas alias x.N.N -> ../Formula/x.rb until x.rb becomes a historical version. It seems sensible to do this also here as it will follow as a versioned formula when 1.19 is out anyway.

**Comment by carlocab** [comment]
Please squash your commits together and use the commit message `envoy@1.17 1.17.3 (new formula)`

**Comment by codefromthecrypt** [comment]
thanks so much for your help @carlocab I will try to pay back by maintaining this

**Comment by codefromthecrypt** [comment]
fyi envoy doesn't work with macOS+arm64 yet due to things including https://github.com/envoyproxy/envoy/pull/16083#issuecomment-823608097

**Review by carlocab — APPROVED** [review]
LGTM. Thanks, @codefromthecrypt. Away from my computer at the moment, so can't trigger a merge. Ping me later this evening (GMT) if I forget.

**Comment by codefromthecrypt** [comment]
cool looking forward to it!
